src/components/PatientRecord/PatientRecord.tsx

```diff
@@ -14,31 +14,133 @@ import {
+  useEffect(() => {
+    if (recordContainerElement?.current && headerElement?.current)
+      recordContainerElement.current.style.height =
+        window.innerHeight - headerElement.current.clientHeight + 'px';
+  }, [isExpanded, headerElement]);
+
+  if (isExpanded) {
+    return (
+      <div className={styles.record} ref={recordContainerElement}>
+        <div className={styles.sidebar}>
+          {resourceTypes.map(resourceType => (
+            <PatientRecordElement resourceType={resourceType} key={resourceType} />
+          ))}
+        </div>
+
+        <div className={styles.recordToggle} onClick={expand}>
+          <FontAwesomeIcon icon={faChevronLeft} />
+        </div>
+      </div>
+    );
+  } else {
+    return (
+      <div className={styles.record}>
+        <div className={styles.recordToggle} onClick={expand}>
+          <FontAwesomeIcon icon={faNotesMedical} />
+        </div>
+      </div>
+    );
+  }
+};
+
+const PatientRecordElement: FC<PatientRecordElementProps> = ({ resourceType }) => {
   const resources = usePatientRecords().patientRecords;
+  const resourcesByType = getResourceByType(resources, resourceType);
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
+
+  const chevron: IconProp = isExpanded ? faChevronUp : faChevronDown;
+  const resourceCount: string =
+    resourceType !== 'Patient' ? '(' + resourcesByType.length + ')' : '';
 
   return (
-    <div>
-      <PatientVisualizer patient={patient} />
-      <ConditionsVisualizer rows={getResourceByType(resources, 'Condition')} />
-      <ObservationsVisualizer rows={getResourceByType(resources, 'Observation')} />
-      <ReportsVisualizer rows={getResourceByType(resources, 'DiagnosticReport')} />
-      <MedicationsVisualizer rows={getResourceByType(resources, 'MedicationRequest')} />
-      <AllergiesVisualizer rows={getResourceByType(resources, 'AllergyIntolerance')} />
-      <CarePlansVisualizer rows={getResourceByType(resources, 'CarePlan')} />
-      <ProceduresVisualizer rows={getResourceByType(resources, 'Procedure')} />
-      <EncountersVisualizer rows={getResourceByType(resources, 'Encounter')} />
-      <ImmunizationsVisualizer rows={getResourceByType(resources, 'Immunization')} />
+    <div className={styles.element}>
+      <div className={styles.title} onClick={(): void => setIsExpanded(!isExpanded)}>
+        <div>
+          {resourceType} {resourceCount}
+        </div>
+        <div className={styles.expand}>
+          <FontAwesomeIcon icon={chevron} />
+        </div>
+      </div>
+
+      {isExpanded && (
+        <div className={styles.elementContainer}>
+          <Visualizer resourceType={resourceType} resourcesByType={resourcesByType} />
+        </div>
+      )}
     </div>
   );
 };
 
+const Visualizer: FC<VisualizerProps> = ({ resourceType, resourcesByType }) => {
+  const patient = usePatient();
+  if (resourceType === 'Patient') return <PatientVisualizer patient={patient} />;
```

Comment:
I won't hold up the PR for just this, but anytime I see a long chain of if-else-if on a single variable I think it should be a `switch` instead. Since everything here is so similar it might also be possible to do a "dynamic type", though not sure if that's possible or easy with TypeScript:
```       
const Something = ...;
return <Something rows={resourcesByType} />
```